core/src/test/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoHashGridTests.java

```diff
@@ -0,0 +1,90 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.search.aggregations.bucket.geogrid;
+
+import org.apache.lucene.index.IndexWriter;
+import org.elasticsearch.common.geo.GeoHashUtils;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.search.aggregations.InternalAggregationTestCase;
+import org.elasticsearch.search.aggregations.InternalAggregations;
+import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class InternalGeoHashGridTests extends InternalAggregationTestCase<InternalGeoHashGrid> {
+
+    @Override
+    protected InternalGeoHashGrid createTestInstance(String name, List<PipelineAggregator> pipelineAggregators,
+                                                     Map<String, Object> metaData) {
+        int size = randomIntBetween(1, 100);
+        List<InternalGeoHashGrid.Bucket> buckets = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            long geoHashAsLong = GeoHashUtils.longEncode(randomInt(90), randomInt(90), 4);
+            buckets.add(new InternalGeoHashGrid.Bucket(geoHashAsLong, randomInt(IndexWriter.MAX_DOCS), InternalAggregations.EMPTY));
+        }
+        return new InternalGeoHashGrid(name, size, buckets, pipelineAggregators, metaData);
+    }
+
+    @Override
+    protected Writeable.Reader<InternalGeoHashGrid> instanceReader() {
+        return InternalGeoHashGrid::new;
+    }
+
+    @Override
+    protected void assertReduced(InternalGeoHashGrid reduced, List<InternalGeoHashGrid> inputs) {
+        Map<Long, List<InternalGeoHashGrid.Bucket>> map = new HashMap<>();
+        for (InternalGeoHashGrid input : inputs) {
+            for (GeoHashGrid.Bucket bucket : input.getBuckets()) {
+                InternalGeoHashGrid.Bucket internalBucket = (InternalGeoHashGrid.Bucket) bucket;
+                List<InternalGeoHashGrid.Bucket> buckets = map.get(internalBucket.geohashAsLong);
+                if (buckets == null) {
+                    map.put(internalBucket.geohashAsLong, buckets = new ArrayList<>());
+                }
+                buckets.add(internalBucket);
+            }
+        }
+        List<InternalGeoHashGrid.Bucket> expectedBuckets = new ArrayList<>();
+        for (Map.Entry<Long, List<InternalGeoHashGrid.Bucket>> entry : map.entrySet()) {
+            long docCount = 0;
+            for (InternalGeoHashGrid.Bucket bucket : entry.getValue()) {
+                docCount += bucket.docCount;
+            }
+            expectedBuckets.add(new InternalGeoHashGrid.Bucket(entry.getKey(), docCount, InternalAggregations.EMPTY));
+        }
+        expectedBuckets.sort((first, second) -> {
+            int cmp = Long.compare(first.docCount, second.docCount) * -1;
```

Comment:
please do `Long.compare(second.docCount, first.docCount)` instead of multiplying by `-1`